Python/modsupport.c

```diff
@@ -88,6 +88,19 @@ static PyObject *do_mklist(const char**, va_list *, char, Py_ssize_t);
 static PyObject *do_mkdict(const char**, va_list *, char, Py_ssize_t);
 static PyObject *do_mkvalue(const char**, va_list *);
 
+static int
+check_end(const char *f, char endchar)
+{
+    while (*f == ',' || *f == ':' || *f == ' ' || *f == '\t') {
+        f++;
+    }
+    if (*f != endchar) {
```

Comment:
For performance, it may be better to test the fast path first.
```suggestion
    if (*f == endchar) {
        return 1;
    }
    while (*f == ',' || *f == ':' || *f == ' ' || *f == '\t') {
        f++;
    }
    if (*f != endchar) {
```